### PR TITLE
http2: an accumulation of another round of edits

### PR DESCRIPTION
**Pull request by jasnell**
I will be updating this incrementally over the next few days. There are multiple general edits in here.

### CODE REVIEW

File: lib/internal/http2/core.js [hunk 1]
```diff
@@ -200,33 +231,89 @@ function onSessionShutdownComplete(status, wrap) {
...
+  if (options.silent === undefined)
+    options.silent = false;
+  else if (typeof options.silent !== 'boolean') {
+    const err = new errors.RangeError('ERR_INVALID_OPT_VALUE',
+                                      'silent',
+                                      options.silent);
+    Error.captureStackTrace(err, validatePriorityOptions);
+    throw err;
+  }
+}
```
Comment (mcollina) [inline lib/internal/http2/core.js:321]: this is not really readable, I would prefer a blank line between the if blocks, and braces everywhere.

File: lib/internal/http2/core.js [hunk 2]
```diff
@@ -332,15 +424,15 @@ class Http2Session extends EventEmitter {
     // streams, these may all need to be made aware of changes
     // in state that occur -- such as when the associated socket
     // is closed. To do so, we need to set the max listener count
-    // to something more reasonable.
+    // to something more reasonable to avoid the memory leak warning.
```
Comment (mcollina) [inline lib/internal/http2/core.js]: avoiding a warning is not newbie friendly, as it encourages a good practice. The actual reason is that we can have an extremely high number of streams multiplexed over the session.

File: lib/internal/http2/core.js [hunk 3]
```diff
@@ -355,7 +447,7 @@ class Http2Session extends EventEmitter {
 
   get destroyed() {
     const state = this[kState];
-    return Boolean(state.destroyed);
+    return !!state.destroyed;
```
Comment (mcollina) [inline lib/internal/http2/core.js]: do we need this check at all? I think we should always have a `destroyed` property in the state.

File: lib/internal/http2/core.js [hunk 4]
```diff
@@ -676,7 +804,7 @@ class Http2Stream extends Duplex {
     req.callback = cb;
     req.oncomplete = afterDoStreamWrite;
     req.async = false;
-    const enc = data instanceof Buffer ? 'buffer' : encoding;
+    const enc = Buffer.isBuffer(data) ? 'buffer' : encoding;
```
Comment (mcollina) [inline lib/internal/http2/core.js]: is this needed? `encoding` should already contain this info.

**Review by mcollina — APPROVED** [review]
Some nits, LGTM after those.

File: lib/internal/http2/core.js (hunk 3 above)
Comment (jasnell) [reply lib/internal/http2/core.js]: likely a good idea

File: lib/internal/http2/core.js (hunk 4 above)
Comment (jasnell) [reply lib/internal/http2/core.js]: yep, you're right :-)

**Comment by jasnell** [comment]
@mcollina ... updated

**Review by mcollina — APPROVED** [review]
LGTM

**Comment by jasnell** [comment]
added some linting fixes and a fix for server-rst-stream test

**Comment by jasnell** [comment]
Rebased and updated. PTAL

**Comment by jasnell** [comment]
CI: https://ci.nodejs.org/job/node-test-pull-request/8049/

(first time running the http2 branch on our CI... fingers crossed)

**Comment by jasnell** [comment]
CI shows a handful of failures that would need to be looked at but nothing specific to this PR. Landing

**Comment by jasnell** [comment]
Landed!

**Pull request closed without merging**